sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -906,6 +906,18 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
                         .map(x -> getFieldSchema(x.getType(), x.getName(), namespace))
                         .collect(Collectors.toList()));
             break;
+
+          case "NVARCHAR":
```

Comment:
Can you please use the JDBCType constants here (see SchemaUtil.java for ref). Also can you please add a test for the VARCHAR and DATE cases. Thanks!

Reply:
JDBCType constants are Integer typed, 
do you recommend me creating a new `JdbcLogicalType` in `org.apache.beam.sdk.schemas.logicalTypes`

Reply:
JdbcType are an enum so you can resolve the name via .getName() as done in https://github.com/apache/beam/blob/79c1932af6df31c677676a43afd6f68406225686/sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java#L143. So we don't need to create the extra class.

Reply:
The proposed change is as part of a switch-case statement, and needs a String constant, so can't use the `[Enum].getName()` method. Sorry, am unable to think of any other way to limit this to AvroUtils only,
the alternative I can think of is making changes upstream in the `SchemaUtil#jdbcTypeToBeamFieldConverter` method, but I fear it may have a much wider impact radius.

Reply:
Oh I see, you are right also The SchemaUtil is in a connector module and this is core, so ok let's deal with it as Strings then. Can you just add the tests for both cases and we should be good to go. Thanks @anantdamle

Reply:
Done.